client.go

```diff
@@ -454,13 +457,13 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 			}
 		}
 
-		if c.RequestLogHook != nil && logger != nil {
+		if c.RequestLogHook != nil {
 			switch v := logger.(type) {
 			case Logger:
 				c.RequestLogHook(v, req.Request, i)
 			case hclog.Logger:
 				c.RequestLogHook(hookLogger{v}, req.Request, i)
-			default:
+			case nil:
```

Comment:
I have a personal preference for being super explicit, but have no problem with changing this to `default` if you prefer that. Same for [below](https://github.com/hashicorp/go-retryablehttp/pull/67/files#diff-d0b7f86cea13f90d3cc0cc2338da8d5fR497)

Reply:
I'd prefer it be default in case someone passes a weird logger it does the same thing.

Reply:
Cool, changed!